### PR TITLE
Absolute paths

### PR DESCRIPTION
**Pull request by samuelhwilliams**
## Summary
It's good practice to use the absolute path to an executable for the sake of clarity and to avoid changes to the PATH being able to accidentally redirect to an alternative executable. Not a massive risk for Docker, but we may as well do it. Also clarify that we're using pip3/python3 rather than the pre-installed python2/pip2 versions.

P.S. This PR sits on top of #28, so until that is merged, please only look at the changes introduced by commit 6eebb43395337290959bdf7eb8f64cc4a428d8b2

### CODE REVIEW

**Comment by samuelhwilliams** [comment]
I got called up on it in my last pentest. :man_shrugging: In reality, probably, _yes_, but why not.

**Comment by benvand** [comment]
The python paths look good as per the writeup i did here:
https://github.com/alphagov/digitalmarketplace-manual/pull/55/files

It feels a bit odd to be reaching inside the venv to specify _`pip3`_. As it's a `python3` venv the `pip` executable should point to `pip3` for us. 
Doing this makes the code less cross compatible/ version compatible for no real gain.
For example in the case of a _(god forbid)_ `python4`.

**Comment by samuelhwilliams** [comment]
That's arguably true, although I don't think it's a big deal. I use `pip3` here because I think it just makes it flat-out clear that this is going to be run using Python3, where we still have various bits of infrastructure/code running on Python2.

File: api.docker [hunk 1]
```diff
@@ -2,9 +2,9 @@ FROM digitalmarketplace/base
 
 COPY nginx/api /etc/nginx/sites-enabled/api
 
-ONBUILD RUN python -m venv venv
+ONBUILD RUN /usr/local/bin/python3 -m venv venv
 ONBUILD COPY requirements.txt ${APP_DIR}
-ONBUILD RUN venv/bin/pip install --no-cache-dir -r requirements.txt
+ONBUILD RUN /app/venv/bin/pip3 install --no-cache-dir -r requirements.txt
 
 ONBUILD COPY . ${APP_DIR}
```
Comment (galund) [inline api.docker]: This seems to also rely on the current directory, so for consistency should we also substitute `/app` for `.`?

**Pull request merged**